### PR TITLE
Make ButtonGroup.Button a wrapper for Button

### PR DESCRIPTION
**Pull request by jmsv6d**
### Summary
ButtonGroup.Button should be a higher order component that wraps terra-button with a restricted API.  #188 

Thanks for contributing to Terra. 
@cerner/terra

[CONTRIBUTORS.md]: ../blob/master/CONTRIBUTORS.md
[License]: ../blob/master/LICENSE

### CODE REVIEW

File: packages/terra-button-group/package.json [hunk 1]
```diff
@@ -27,11 +27,13 @@
   "peerDependencies": {
     "react": "^15.4.2",
     "react-dom": "^15.4.2",
-    "terra-mixins": "^1.0.0"
+    "terra-mixins": "^1.0.0",
+    "terra-button": "^0.1.0"
```
Comment (Matt-Butler) [inline packages/terra-button-group/package.json]: For components < 1.x, we have been versioning it like this:
```
"terra-button": "^0.x"
```
Which allows terra-button 0.1, 0.2.

This site is really useful for learning more about npm versioning:
https://semver.npmjs.com/

File: packages/terra-button-group/package.json [hunk 2]
```diff
@@ -27,11 +27,13 @@
   "peerDependencies": {
     "react": "^15.4.2",
     "react-dom": "^15.4.2",
-    "terra-mixins": "^1.0.0"
+    "terra-mixins": "^1.0.0",
+    "terra-button": "^0.x"
```
Comment (StephenEsser) [inline packages/terra-button-group/package.json]: It looks like this missed was missed during the terra-base PR: https://github.com/cerner/terra-core/pull/183

I'd sync up with @bjankord to see if this should have a peerDependency of terra-base, it can probably be added in a different pull request if it doesn't really relate to this one
